include/swift/Basic/Located.h

```diff
@@ -0,0 +1,39 @@
+//===--- Located.h - Source Location and Associated Value ----------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file forward declares and imports various common LLVM datatypes that
+// swift wants to use unqualified.
+//
+//===----------------------------------------------------------------------===//
+
+
+#ifndef SWIFT_BASIC_LOCATED_H
+#define SWIFT_BASIC_LOCATED_H
+#include "swift/Basic/SourceLoc.h"
+
+namespace swift {
+
+template<typename T>
+struct Located {
+
+  T item;
+
+  SourceLoc loc;
```

Comment:
Could you add some doc comments here for the fields and the type? For example [you don't need to stick to this wording if you have better ideas :)]

```
/// A currency type for keeping track of items which were found in the source code.
///
/// Several parts of the compiler need to keep track of a `SourceLoc` corresponding to an item, in case
/// they need to report some diagnostics later. For example, the ClangImporter needs to keep track
/// of where imports were originally written. Located makes it easy to do so while making the code
/// more readable, compared to using `std::pair`.
template<typename T>
 struct Located {

   /// The main item whose source location is being tracked.
   T item;

   /// The original source location from which the item was parsed.
   SourceLoc loc;
```

Reply:
Sure. Updated with documentation.